src/UniqueKey/Service.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace SilverStripe\UniqueKey;
+
+use SilverStripe\Core\Injector\Injectable;
+use SilverStripe\ORM\DataObject;
+
+class Service
```

Comment:
Add some PHPDoc

Reply:
Define an interface so other people who would want to override this have something to code against.

Reply:
I also really object to the name `Service` here. It doesn't really mean anything. What about `UniqueKeyGenerator` (provided the namespace is changing to `DataObject`).